### PR TITLE
Login Page

### PR DESCRIPTION
**Pull request by foolhardy21**
## Login Page

This PR has login page components, input validation and theme toggle.

- Files to review:
  - [ ] src/components/Login/Login.js
  - [ ] src/components/Login/LoginCard.js
  - [ ] src/components/Login/LoginForm.js
  - [ ] src/components/Login/LoginHeader.js
  - [ ] src/context/theme-context.js
  - [ ] src/utils/inputValidation.js
  - [ ] src/utils/regExp.js
  - [ ] src/utils/theme-util.js

- Use this link for preview - https://deploy-preview-1--pensive-perlman-311072.netlify.app/

- Screenshots:
d-1
d-2

Thanks.

### CODE REVIEW

File: src/components/Login/LoginForm.js [hunk 1]
```diff
@@ -63,15 +66,15 @@ const LoginForm = () => {
...
 
                 </div>
 
-                <Button onClick={forgotPassword} text='forgot your password?' classes='btn-txt txt-primary txt-md txt-cap bg-primary pd-xs mg-left-xs' />
+                <Button onClick={forgotPassword} text='forgot your password?' classes={`btn-txt ${getTextColor(theme)} ${getBgColor(theme)} txt-md txt-cap pd-xs mg-left-xs`} />
 
             </div>
 
-            <Button onClick={logInUser} text='log in' classes='btn-solid txt-secondary bg-secondary txt-md txt-ucase pd-xs' />
+            <Button onClick={logInUser} text='log in' classes={`btn-solid ${getSolidBtnTextColor(theme)} ${getSolidBtnBgColor(theme)} txt-md txt-ucase pd-xs`} />
```
Comment (VLeads) [inline src/components/Login/LoginForm.js]: Appreciate this button component. Passing everything as a prop here .

File: src/components/Login/LoginHeader.js [hunk 2]
```diff
@@ -2,23 +2,29 @@ import Header from "../Reusable/Header"
...
+import { getTextColor } from "../../utils/theme-util"
 
 const LoginHeader = () => {
+    const { theme, toggleTheme } = useTheme()
 
     return (
         <Header>
 
-            <Text classes='txt-ucase txt-primary txt-lg' text='sneakerstore' />
+            <Text classes={`txt-ucase ${getTextColor(theme)} txt-primary txt-lg`} text='sneakerstore' />
```
Comment (VLeads) [inline src/components/Login/LoginHeader.js]: Nicely done. Everything is converted to a Component even "Text".  But you may get into issue in future. If you have to style text differently eg. including bold text somewhere, increasing fontsize of a particular character etc.

**Review by VLeads — APPROVED** [review]
Eveything is done nicely. You may look at suggestion I have given. Rest all looks good.

File: src/components/Login/Login.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import { useTheme } from "../../context/theme-context"
+import { getBgColor } from "../../utils"
+
+const Login = () => {
+    const { theme } = useTheme()
+
+    return (
+        <div style={{
+            minHeight: '100vh',
+        }} className={`${getBgColor(theme)}`}>
```
Comment (Maianki) [inline src/components/Login/Login.js:12]: If possible avoid inline css.

File: src/components/Reusable/Icon.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { useTheme } from "../../context/theme-context"
+import { getIconColor } from '../../utils'
+
+const Icon = (props) => {
+    const { theme } = useTheme()
+
+    return (
+        <span id={props.id} className={`material-icons ${getIconColor(theme)} ${props.classes}`} >
+            {props.children}
```
Comment (Maianki) [inline src/components/Reusable/Icon.js]: Nitpicking : You can use destructuring here.

File: src/components/Reusable/Input.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { useTheme } from "../../context/theme-context"
+import { getBgColor, getTextColor } from "../../utils"
+
+const Input = (props) => {
+    const { theme } = useTheme()
+
+    return (
+        <input type={props.type} placeholder={props.placeholder} value={props.value} onChange={props.onChange}
+            className={`${getBgColor(theme)} ${getTextColor(theme)} ${props.classes}`} />
+    )
```
Comment (Maianki) [inline src/components/Reusable/Input.js]: Here too you can destructure props object.

**Review by Maianki — APPROVED** [review]
Great work Vinay.
Loved that you have made separate reusable components for buttons, input, and icons too.
Your code is very readable.
Small suggestion try destructuring props it will enhance readability.

File: src/components/Login/LoginHeader.js (hunk 2 above)
Comment (foolhardy21) [reply src/components/Login/LoginHeader.js]: thanks @VLeads. For the case mentioned, I have dedicated classes in my component library. So I can pass those classes to make the changes.

File: src/components/Login/Login.js (hunk 3 above)
Comment (foolhardy21) [reply src/components/Login/Login.js:12]: sure. ill make the changes.

File: src/components/Reusable/Icon.js (hunk 4 above)
Comment (foolhardy21) [reply src/components/Reusable/Icon.js]: thanks @Maianki . i have made the changes.

File: src/components/Reusable/Input.js (hunk 5 above)
Comment (foolhardy21) [reply src/components/Reusable/Input.js]: i have made the changes. thanks

**Pull request merged**